pinot-controller/src/main/java/org/apache/pinot/controller/api/events/MetadataEventNotifierFactory.java

```diff
@@ -27,16 +28,19 @@ public abstract class MetadataEventNotifierFactory {
   public static final Logger LOGGER = LoggerFactory.getLogger(MetadataEventNotifierFactory.class);
   public static final String METADATA_EVENT_CLASS_CONFIG = "factory.class";
 
+  protected static PinotHelixResourceManager _pinotHelixResourceManager;
+
   public abstract void init(PinotConfiguration configuration);
```

Comment:
If you need `PinotHelixResourceManager` in the `MetadataEventNotifierFactory`, shouldn't you add it into the `init()`? Putting it as static variable is quite hacky

Reply:
Modified the code